### ISSUE TITLE
Breaking changes information missing from ChangeLog

### ISSUE THREAD

**Issue by nickb937** [SeaQL/sea-orm#2414]
**Upgrading from SeaORM 0.12 to 1.1**

There seem to be some changes that are breaking changes but aren't mentioned in the Changelog, which tripped me up during an upgrade.  It would be helpful to expand the information in the ChangeLog for others.

1. The requirement to always define an `ActiveModel` for a table.   In v0.12 I did not implement `ActiveModel` for some tables because they are read-only, managed directly by some database triggers.  In this instance SeaORM defines the schema and I wanted a compile error should somebody try to write code to update the rows.

```rust
error[E0412]: cannot find type `ActiveModel` in this scope
 --> table.rs:5:39
  |
5 | #[derive(Copy, Clone, Default, Debug, DeriveEntity)]
  |                                       ^^^^^^^^^^^^
  |
  = note: this error originates in the derive macro `DeriveEntity` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider importing one of these items
```

I see it's now written in the doc here that it should always be defined: https://www.sea-ql.org/SeaORM/docs/generate-entity/entity-structure/

2. `ColumnType::String` definitions seem to have changed, wrapped by StringLen.
ColumnType::String(Some(32)).def()

3. `use sea_orm::sea_query::BlobSize` has been Dropped, but no indication of what it should have been replaced with. 

I guess that `ColumnType::Binary(BlobSize::Blob(None)).def()` should be `ColumnType::Binary(0).def()`

ColumnType defines:

```rust
    Binary(u32),
    VarBinary(StringLen),
```

On PostgreSQL should I now be using ` ColumnType::VarBinary(StringLen::None)` or `ColumnType::Binary(0).def()` or does it matter?

4. Should we be changing `DeriveEntity` or `DeriveEntityModel` ?  Is there any difference between the two? Is `DeriveEntity` deprecated?

```rust
#[derive(Copy, Clone, Default, Debug, DeriveEntity)]
pub struct Entity;
```

https://github.com/SeaQL/sea-orm/blob/5f1a9d2a46c9b0776c57311782e72dab8e5c6948/src/entity/mod.rs#L30

-vs-

https://www.sea-ql.org/SeaORM/docs/generate-entity/entity-structure/

**Comment by tyt2y3** [comment]
Yes, I think I changed the derive macro in https://github.com/SeaQL/sea-orm/pull/2186 I know that is breaking, and it should only impact relatively few users. we've never thought about "entity without active model = read only" use case. and may be we can work something out to make it a feature.

I am not sure if it will work, can a trait associated constant, say `READ_ONLY = true` in `EntityTrait`, trigger a compile-time error with a `const fn`?

regarding binary, `BlobSize` is really a MySQL thing, it doesn't exist in Postgres, usually `VarBinary` is used.
